plugins/shutdown.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const joi = require('joi');
+const logger = require('screwdriver-logger');
+
+const tasks = {};
+const taskSchema = joi.object({
+    taskname: joi.string().required(),
+    task: joi.func().required(),
+    timeout: joi.number().integer()
+});
+
+/**
+ * Function to return promise timeout or resolution
+ * whichever happens first
+ * @param {function} fn
+ * @param {string} timeout
+ */
+function promiseTimeout(fn, timeout) {
+    return Promise.race([
+        Promise.resolve(fn),
+        new Promise(resolve => {
+            setTimeout(() => {
+                resolve(`Promise timed out after ${timeout} ms`);
+            }, timeout);
+        })
+    ]);
+}
+
+/**
+ * Hapi plugin to handle serve graceful shutdown
+ * @method register
+ * @param  {Hapi.Server}    server
```

Comment:
[nits]
Is lint working?  It looks like there are many space between {Hapi.Server} and server

Reply:
Lint may not be working. I deleted the space.